connect-inject/container_init.go

```diff
@@ -124,6 +133,9 @@ services {
     upstreams {
       destination_name = "{{ .Name }}"
       local_bind_port = {{ .LocalPort }}
+			{{ if .Datacenter }}
+			datacenter = "{{ .Datacenter }}"
+			{{ end }}
```

Comment:
```suggestion
      {{- end }}
```
Similarly removes the blank line that would otherwise be inserted here and updates indentation.